### DISCUSSION TITLE
Reaching out for mentors for project ,3. Desktop Chat-Bot application

### DISCUSSION THREAD

**Discussion by BlossomBreeze25** [openvinotoolkit/openvino#23207] [Google Summer of Code]
Respected sir / ma'am,
I am Ragini Devi Thakkar , 4th Semester BCA(AI) student, want to join your organization. i have knowledge about AI and languages like python , java . html , css . i am interested in this project (3. Desktop Chat-Bot application)and I would love to contribute to your organization but could you please tell me how to get started?

Hoping to hear from you soon.

Regards,

Ragini Devi Thakkar.

**Comment by adrianboguszewski** [comment]
@vishniakov-nikolai @almilosz please look at it

**Comment by adismort14** [comment]
Greetings Mentors! :wave: 

This is Aditya, junior year Computer Science undergrad @ IIT Bhilai. 
Have a couple questions regarding the project. Will keep it short to respect your time.

0] Can we use pre-trained Language Models and then fine-tune them for our task, if any?
1] Related to the above point, is there any specific use-case in mind currently? Will look in that direction too if there is.
2] Any specification/constraint for the LLM such as must be lightweight etc.

It would be appreciated if I could be pointed to any **related open issues** that need to be fixed. This would help me familiarise myself with the codebase too.

The work I am doing at my professors Research Lab aligns with this project and would love to contribute to OpenVINO.
Thanks! :smile: 

cc: @vishniakov-nikolai , @almilosz, @adrianboguszewski

**Reply by almilosz** [reply]
Hello Aditya,
Thank you for your interest in implementing _Desktop_ Chat-Bot _application._
Here you can find the description of the [task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#3-desktop-chat-bot-application)

0] You can use either fine-tuned or just pre-trained models
1] There are no restrictions on the use-case. Your chat-bot can be specialized in one area.
2) Here please remember that the final goal is to prepare desktop app that will work without access to the internet 

It's great that you want to fix some open issue! We kindly request all participants to review the [Prerequisite task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task).
I can assign you to [this issue](https://github.com/openvinotoolkit/openvino/issues/23441) or you can find something yourself [here](https://github.com/orgs/openvinotoolkit/projects/3). When you decide, please leave a comment under the issue ;)

I will be glad to hear more about your other projects.
Alicja

**Reply by adismort14** [reply]
Thanks a lot for your prompt and helpful reply Alicja! :smile: 

Yes, the issue #23441 should be doable and I will comment under it. :+1: 

The project I am working on mainly involves 5G network troubleshooting chatbot using RAG technique and a fine tuned LLM. The dataset corpus that we are vectorising is roughly 30 GBs of docs (entire 3GPP specs) and to get the optimal responses we are looking into generating the graph to link internal and external references between these documents. 

Well, that's the gist of it. Would love to dive deeper and brief my other projects too but for now some OpenVino issues need fixing and I am up for it :)

Aditya.

**Reply by vishniakov-nikolai** [reply]
Hi @adismort14! Please, pay attention that to participate in GSoC you have to submit your application throw GSoC site https://summerofcode.withgoogle.com/register/contributor

I've seen your submission, but to participate in Desktop Chat-Bot application development you need send your application and noticed your wish to participate in certain project.
